flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -1494,6 +1519,22 @@ class CodeGenerator(
   }
 
   /**
+    * Adds a reusable [[java.util.Random]] to the member area of the generated [[Function]].
+    *
+    * @return member variable term
+    */
+  def addReusableRandom(): String = {
+    val fieldTerm = newName("random")
+
+    val field =
+      s"""
+         |transient java.util.Random $fieldTerm = null;
```

Comment:
`Random` implements `Serializable`. I think we initialize `Random` already here, remove `transient` and add `final`.